runtime/vm/CRIUHelpers.cpp

```diff
@@ -413,8 +413,26 @@ runDelayedLockRelatedOperations(J9VMThread *currentThread)
 	Assert_VM_true(vm->checkpointState.checkpointThread == currentThread);
 
 	while (NULL != delayedLockingOperation) {
-		omrthread_monitor_t monitorPtr = NULL;
 		j9object_t instance = J9_JNI_UNWRAP_REFERENCE(delayedLockingOperation->globalObjectRef);
+		omrthread_monitor_t monitorPtr = NULL;
+		if (J9_SINGLE_THREAD_MODE_OP_INTERRUPT == delayedLockingOperation->operation) {
+			J9VMThread *targetThread = J9VMJAVALANGTHREAD_THREADREF(currentThread, instance);
```

Comment:
can we add a common helper for this, so the code isnt dplicated?

Reply:
Added `VM_VMHelpers::threadInterruptImpl`.